### PR TITLE
Add region handling to UpdateModelConfigDefaultValues

### PR DESCRIPTION
**Pull request by reedobrien**
Refs: model-config-tree

QA: Unit tests all pass.  This should have gone in with the rest of the backend model-config region bits. But it didn't it is required for the cli bits to work, so hoping to merge this so we can get feedback on the CLI bits.

### CODE REVIEW

File: state/modelconfig.go [hunk 1]
```diff
@@ -134,8 +135,18 @@ func (st *State) modelConfigValues(modelCfg attrValues) (config.ConfigValues, er
 }
 
 // UpdateModelConfigDefaultValues updates the inherited settings used when creating a new model.
-func (st *State) UpdateModelConfigDefaultValues(attrs map[string]interface{}, removed []string) error {
-	settings, err := readSettings(st, globalSettingsC, controllerInheritedSettingsGlobalKey)
+func (st *State) UpdateModelConfigDefaultValues(attrs map[string]interface{}, removed []string, cloudTag, regionName string) error {
```
Comment (axw) [inline state/modelconfig.go]: at this layer, please either pass in a cloud *name*, or a names.CloudTag

File: state/modelconfig.go [hunk 2]
```diff
@@ -134,8 +135,18 @@ func (st *State) modelConfigValues(modelCfg attrValues) (config.ConfigValues, er
...
+	var key string
+	if cloudTag != "" && regionName != "" {
+		cloud, err := names.ParseCloudTag(cloudTag)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		key = regionSettingsGlobalKey(cloud.Id(), regionName)
+	} else {
```
Comment (axw) [inline state/modelconfig.go]: return an error if `cloudTag != "" && regionName == ""` ?

Comment (reedobrien) [inline state/modelconfig.go]: Changed the machinery to use environs.RegionSpec to define the cloud/region and added a constructor that barfs if cloud or region is an empty string.

File: state/modelconfig.go (hunk 1 above)
Comment (reedobrien) [inline state/modelconfig.go]: fixed

File: apiserver/modelmanager/modelmanager.go [hunk 3]
```diff
@@ -839,9 +847,35 @@ func (c *ModelManagerAPI) UnsetModelDefaults(args params.UnsetModelDefaults) (pa
...
+			if err != nil {
+				results.Results[i].Error = common.ServerError(
+					errors.Trace(err))
+				continue
+			}
+			results.Results[i].Error = common.ServerError(
+				c.state.UpdateModelConfigDefaultValues(nil, arg.Keys, rspec),
+			)
+		} else {
+			results.Results[i].Error = common.ServerError(
```
Comment (axw) [inline apiserver/modelmanager/modelmanager.go]: I think it'd be a little neater to de-dupe these two calls, and do the conditional rspec thing like you did in the other method.

Comment (reedobrien) [inline apiserver/modelmanager/modelmanager.go]: Good idea. Fixed. Sqaushing for merge.

**Comment by reedobrien** [comment]
$$merge$$

**Comment by reedobrien** [comment]
$$merge$$

**Pull request merged**